### PR TITLE
Email competition

### LINKED ISSUE #276: Option: Participate in Competition at Fund setup
For initial competition, adding email field. Eg after Portfolio name
screenshot from 2017-06-26 11-59-54

### PR DESCRIPTION
**Pull request by jennazenk**
closes #276

### CODE REVIEW

**Comment by retotrinkler** [comment]
good

File: imports/api/users.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  Users.insert({
+    email,
+    address,
+    verified: false,
+  });
+};
+
+Meteor.methods({
+  'users.add': (userInfo) => {
+    check(userInfo.email, String);
```
Comment (schmidsi) [inline imports/api/users.js]: Really check if it is a valid email address. The client-side check is not sufficient. It's easy to circumvent this check and insert bad emails into our system.

Example RegExp:
/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/

Or use other library like [validator](https://www.npmjs.com/package/validator)

File: imports/redux/middlewares/user.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      });
+      break;
+    }
+    case web3Types.UPDATE: {
+      const currentAccount = currentState.web3.account;
+      const updatedAccount = action.account;
+      if (currentAccount !== updatedAccount) {
+        const user = Users.findOne({ address: updatedAccount });
+        if (user) {
+          store.dispatch(creators.updateIsRegistered(true));
```
Comment (schmidsi) [inline imports/redux/middlewares/user.js]: It's better to only dispatch one action at the time. I.e. `creators.updateVerification({ registered: true, verified: false })`

File: imports/ui/components/portal/portalNew.js [hunk 3]
```diff
@@ -41,10 +41,16 @@ Template.portalNew.events({
     event.preventDefault();
     Version.setProvider(web3.currentProvider);
 
+    // if (!templateInstance.find('input#email').value) {
```
Comment (schmidsi) [inline imports/ui/components/portal/portalNew.js]: Why is this commented out?
If not used anymore: delete
If used: activate again

File: imports/ui/components/ux/uxPages.js [hunk 4]
```diff
@@ -32,3 +39,25 @@ Template.uxServerConnection.helpers({});
...
+    const currentState = store.getState().user;
+    template.isRegistered.set(currentState.isRegistered);
+  });
+});
+
+Template.uxInsufficientFunds.helpers({
+  unregistered: () => !Template.instance().isRegistered.get(),
+});
+Template.uxInsufficientFunds.events({
+  'submit form#email': (event, templateInstance) => {
```
Comment (schmidsi) [inline imports/ui/components/ux/uxPages.js]: #60: Use `.js-` selectors

**Pull request merged**